src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -824,6 +828,10 @@ private void DetectAttributeAndOptionConflicts(DiagnosticBag diagnostics)
                     }
                 }
             }
+            if (_compilation.Options.PublicSign && (object)this.AssemblyKeyFileAttributeSetting != (object)CommonAssemblyWellKnownAttributeData.StringMissingValue)
```

Comment:
Can you move the check for KeyContainerAttribute down here and combine the error reporting?